packages/axelar-wasm-std/src/msg_id/starknet_field_element_event_index.rs

```diff
@@ -0,0 +1,284 @@
+use core::fmt;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use cosmwasm_std::HexBinary;
+use error_stack::Report;
+use lazy_static::lazy_static;
+use regex::Regex;
+use serde_with::DeserializeFromStr;
+use starknet_types_core::felt::Felt;
+
+use super::Error;
+use crate::nonempty;
+use crate::utils::does_felt_overflow_from_str;
+
+#[derive(Debug, DeserializeFromStr, Clone)]
+pub struct FieldElementAndEventIndex {
+    pub tx_hash: Felt,
+    pub event_index: u64,
+}
+
+impl FieldElementAndEventIndex {
+    pub fn tx_hash_as_hex(&self) -> nonempty::String {
+        format!("0x{}", self.tx_hash_as_hex_no_prefix())
+            .try_into()
+            .expect("failed to convert tx hash to non-empty string")
+    }
+
+    pub fn tx_hash_as_hex_no_prefix(&self) -> nonempty::String {
+        HexBinary::from(self.tx_hash.to_bytes_be())
+            .to_hex()
+            .to_string()
+            .try_into()
+            .expect("failed to convert tx hash to non-empty string")
+    }
+
+    pub fn new<T: Into<Felt> + FromStr>(
+        tx_id: T,
+        event_index: impl Into<u64>,
+    ) -> Result<Self, Error> {
+        Ok(Self {
+            tx_hash: tx_id.into(),
+            event_index: event_index.into(),
+        })
+    }
+}
+
+// A valid field element is max 252 bits, meaning max 63 hex characters after 0x.
+// We require the hex to be 64 characters, meaning that it should be padded with zeroes in order
+// for us to consider it valid.
+const PATTERN: &str = "^(0x[0-9a-f]{64})-(0|[1-9][0-9]*)$";
```

Comment:
where is the 0 padding in this regex?

Reply:
It should come padded and we just check that the length is 64 characters.
It can be padded with 1 or 2 zeroes. If it's not padded - it'll be invalid regex. If they're not zeroes (> 252 bits) the overflow check would fail.